catkin_ws/src/state_estimation/src/state_aggregator.py

```diff
@@ -29,8 +33,20 @@ def update_state(_):
             break
     for sensor in sensor_priorities["z"]:
         if sensor.isActive():
-            z = sensor.z
-            break
+            if sensor.sensor_name == "depth_sensor":
+                ds_z = sensor.z
+                if dvl_z is not None:
+                    offset = math.abs(ds_z - dvl_z)
+                    i += 1
+                    ds_z -= math.exp(-i)*offset
```

Comment:
choice of exp() is good, just make sure to remember to reset i to 0 when the depth sensor turns off! Also same problem here as with the DVL with the offset always being re-calculated